CI: `ethtool.TestIsVirtualDriver` fails in RuntimeDatapathPrivilegedUnitTests

### Test Name

```shell
RuntimeDatapathPrivilegedUnitTests
```


### Failure Output

```shell
failed to check for veth driver for "dummy0": no such device
```


### Stack Trace

```shell
/home/jenkins/workspace/Cilium-PR-Runtime-kernel-net-next/runtime-gopath/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:515
Failed to run privileged unit tests
Expected command: bash -c 'sudo make -C /home/vagrant/go/src/github.com/cilium/cilium/ tests-privileged NO_COLOR=1 | ts "[%H:%M:%S]"; exit "${PIPESTATUS[0]}"' 
To succeed, but it failed:
Exitcode: 2 
Err: Process exited with status 2
Stdout:
 	 [07:45:58] make: Entering directory '/home/vagrant/go/src/github.com/cilium/cilium'
	 [07:46:08] PRIVILEGED_TESTS=true PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/vagrant/go/src/github.com/cilium/cilium/bpf CGO_ENABLED=0 go test  -mod=vendor -tags=osusergo -ldflags "-X github.com/cilium/cilium/pkg/kvstore.consulDummyAddress=https://consul:8443 -X github.com/cilium/cilium/pkg/kvstore.etcdDummyAddress=http://etcd:4002 -X github.com/cilium/cilium/pkg/datapath.DatapathSHA256=e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855" \
	 [07:46:08] 	./... -test.v -timeout 600s  -json -covermode=count -coverprofile=coverage.out | tparse -follow

...

	 [07:49:00] === RUN   TestIsVirtualDriver
	 [07:49:00]     ethtool_linux_test.go:28: IsVirtualDriver("enp0s3") = false
	 [07:49:00]     ethtool_linux_test.go:28: IsVirtualDriver("enp0s8") = false
	 [07:49:00]     ethtool_linux_test.go:28: IsVirtualDriver("enp0s9") = false
	 [07:49:00]     ethtool_linux_test.go:28: IsVirtualDriver("enp0s10") = false
	 [07:49:00]     ethtool_linux_test.go:28: IsVirtualDriver("docker0") = false
	 [07:49:00]     ethtool_linux_test.go:28: IsVirtualDriver("veth8735703") = true
	 [07:49:00]     ethtool_linux_test.go:28: IsVirtualDriver("veth6f0a335") = true
	 [07:49:00]     ethtool_linux_test.go:28: IsVirtualDriver("veth024914e") = true
	 [07:49:00]     ethtool_linux_test.go:28: IsVirtualDriver("cilium_vxlan") = false
	 [07:49:00]     ethtool_linux_test.go:28: IsVirtualDriver("cilium_net") = true
	 [07:49:00]     ethtool_linux_test.go:28: IsVirtualDriver("cilium_host") = true
	 [07:49:00]     ethtool_linux_test.go:28: IsVirtualDriver("instance1") = false
	 [07:49:00]     ethtool_linux_test.go:28: IsVirtualDriver("instance2") = false
	 [07:49:00]     ethtool_linux_test.go:28: IsVirtualDriver("cilium-bgp") = false
	 [07:49:00]     ethtool_linux_test.go:26: failed to check for veth driver for "dummy0": no such device
	 [07:49:00] --- FAIL: TestIsVirtualDriver (0.01s)
	 [07:49:00] FAIL
	 [07:49:00] 	github.com/cilium/cilium/pkg/datapath/linux/ethtool	coverage: 87.5% of statements
	 [07:49:00] FAIL	github.com/cilium/cilium/pkg/datapath/linux/ethtool	0.009s
```


### Standard Output

```shell
-
```


### Standard Error

```shell
07:45:57 STEP: Running BeforeAll block for EntireTestsuite RuntimeDatapathPrivilegedUnitTests
FAIL: Failed to run privileged unit tests
Expected command: bash -c 'sudo make -C /home/vagrant/go/src/github.com/cilium/cilium/ tests-privileged NO_COLOR=1 | ts "[%H:%M:%S]"; exit "${PIPESTATUS[0]}"' 
To succeed, but it failed:
Exitcode: 2 
Err: Process exited with status 2
Stdout:
 	 [07:45:58] make: Entering directory '/home/vagrant/go/src/github.com/cilium/cilium'
	 [07:46:08] PRIVILEGED_TESTS=true PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/vagrant/go/src/github.com/cilium/cilium/bpf CGO_ENABLED=0 go test  -mod=vendor -tags=osusergo -ldflags "-X github.com/cilium/cilium/pkg/kvstore.consulDummyAddress=https://consul:8443 -X github.com/cilium/cilium/pkg/kvstore.etcdDummyAddress=http://etcd:4002 -X github.com/cilium/cilium/pkg/datapath.DatapathSHA256=e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855" \
	 [07:46:08] 	./... -test.v -timeout 600s  -json -covermode=count -coverprofile=coverage.out | tparse -follow
```


### Resources

- Jenkins URL: https://jenkins.cilium.io/job/Cilium-PR-Runtime-kernel-net-next/432/


### Anything else?

_No response_

--- a/pkg/datapath/linux/ethtool/ethtool_linux_test.go
+++ b/pkg/datapath/linux/ethtool/ethtool_linux_test.go
@@ -4,27 +4,49 @@
 package ethtool
 
 import (
-	"errors"
+	"runtime"
 	"testing"
 
 	"github.com/vishvananda/netlink"
-	"golang.org/x/sys/unix"
+	"github.com/vishvananda/netns"
+
+	"github.com/cilium/cilium/pkg/testutils"
 )
 
 func TestIsVirtualDriver(t *testing.T) {
-	links, err := netlink.LinkList()
-	if err != nil {
-		t.Fatalf("failed to get link list: %v", err)
-	}
+	testutils.PrivilegedTest(t)
 
-	for _, link := range links {
-		name := link.Attrs().Name
-		isVirtual, err := IsVirtualDriver(name)
-		if errors.Is(err, unix.EOPNOTSUPP) {
-			continue
-		} else if err != nil {
-			t.Fatalf("failed to check for veth driver for %q: %v", name, err)
-		}
-		t.Logf("IsVirtualDriver(%q) = %t", name, isVirtual)
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	oldns, err := netns.Get()
+	if err != nil {
+		t.Fatalf("failed to get current netns: %v", err)
+	}
+	defer oldns.Close()
+
+	newns, err := netns.New()
+	if err != nil {
+		t.Fatalf("failed to create new netns: %v", err)
+	}
+	defer newns.Close()
+	defer netns.Set(oldns)
+
+	name := "veth0"
+	veth := &netlink.Veth{
+		LinkAttrs: netlink.LinkAttrs{Name: name},
+		PeerName:  "veth1",
+	}
+	err = netlink.LinkAdd(veth)
+	if err != nil {
+		t.Fatalf("failed to create veth link: %v", err)
+	}
+	defer netlink.LinkDel(veth)
+
+	isVirtual, err := IsVirtualDriver(name)
+	if err != nil {
+		t.Fatalf("error checking veth link %q: %v", name, err)
+	} else if !isVirtual {
+		t.Errorf("IsVirtualDriver(%q) = %t, want true", name, isVirtual)
 	}
 }

